server/chat/backend/agent/tools/cloud_tools.py

```
@@ -1382,6 +1382,23 @@ def cloud_exec_wrapper(provider: str, command: str, output_file: Optional[str] =
                 args_schema=GetConnectedReposArgs
             )
         elif name == 'github_rca':
+            # Wrap to always inject incident_time from the authoritative DB timestamp
+            # stored in state. The field is not in GitHubRCAArgs so the agent LLM
+            # never supplies a value — the wrapper is the only source.
+            def _github_rca_pinned(*args: Any, _inner: Any = final_func, **kwargs: Any) -> Any:
```

Ruff (0.15.13)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `*args`

(ANN401)